### ISSUE TITLE
Game crashes when trying to save or load plan

### ISSUE THREAD

**Issue by Yunichi** [Kamikaze94/WolfHUD#865]
The new update has removed the option to save and load plan, but it's still there on Wolfhud. If you try to save or load, it crashes.

**Comment by BuckMyAsh** [comment]
Can confirm it crashes every time on Shadow Raid. Crash file:
Application has crashed: C++ exception
mods/WolfHUD-master/lua/PrePlanManager.lua:642: attempt to call method '_get_type_cost' (a nil value)



SCRIPT STACK

saved_assets_name() @mods/WolfHUD-master/lua/PrePlanManager.lua:624
@mods/WolfHUD-master/lua/PrePlanManager.lua:174
modify_func() lib/managers/menumanager.lua:7413
get_node() core/lib/managers/menu/coremenudata.lua:36
get_node() core/lib/managers/menu/coremenulogic.lua:310
core/lib/managers/menu/coremenulogic.lua:77
_execute_action_queue() core/lib/managers/menu/coremenulogic.lua:54
update() core/lib/managers/menu/coremenulogic.lua:64
update() core/lib/managers/menu/coremenumanager.lua:155
update() lib/managers/menumanagerpd2.lua:20
original() lib/setups/setup.lua:858
update() @mods/base/req/core/Hooks.lua:188
update() lib/setups/gamesetup.lua:718
update() lib/setups/networkgamesetup.lua:21
original() core/lib/setups/coresetup.lua:557
@mods/base/req/core/Hooks.lua:265


-------------------------------

Callstack:

         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     zip_get_name                                        
                           ???  (???)     ???                                                 
                         ntdll  (???)     RtlDosSearchPath_Ustr                               
                         ntdll  (???)     RtlFreeHeap                                         
                         ntdll  (???)     RtlSetLastWin32Error                                
                    KERNELBASE  (???)     CreateFileW                                         
                    KERNELBASE  (???)     CreateFileW                                         
                         ntdll  (???)     ZwQueryVirtualMemory                                
                    KERNELBASE  (???)     VirtualQueryEx                                      
                    KERNELBASE  (???)     SetUnhandledExceptionFilter                         
                    KERNELBASE  (???)     SetUnhandledExceptionFilter                         
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     zip_get_name                                        


-------------------------------

Current thread: Main

-------------------------------

**Comment by K1R1Y4** [comment]
Same thing happened to me everytime

Thu Aug 13 14:02:59 2020


Application has crashed: C++ exception
mods/WolfHUD-master/lua/PrePlanManager.lua:637: attempt to call method '_get_type_cost' (a nil value)



SCRIPT STACK

saved_assets_name() @mods/WolfHUD-master/lua/PrePlanManager.lua:624
@mods/WolfHUD-master/lua/PrePlanManager.lua:174
modify_func() lib/managers/menumanager.lua:7413
get_node() core/lib/managers/menu/coremenudata.lua:36
get_node() core/lib/managers/menu/coremenulogic.lua:310
core/lib/managers/menu/coremenulogic.lua:77
_execute_action_queue() core/lib/managers/menu/coremenulogic.lua:54
update() core/lib/managers/menu/coremenulogic.lua:64
update() core/lib/managers/menu/coremenumanager.lua:155
update() lib/managers/menumanagerpd2.lua:20
original() lib/setups/setup.lua:858
update() @mods/base/req/core/Hooks.lua:188
update() lib/setups/gamesetup.lua:718
update() lib/setups/networkgamesetup.lua:21
core/lib/setups/coresetup.lua:557


-------------------------------

Callstack:

         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     zip_get_name                                        
                         ntdll  (???)     RtlDosSearchPath_Ustr                               
                         ntdll  (???)     RtlSetLastWin32Error                                
                         ntdll  (???)     RtlAllocateHeap                                     
                         ntdll  (???)     LdrCallEnclave                                      
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     NvOptimusEnablement                                 
         payday2_win32_release  (???)     ???                                                 
                           ???  (???)     ???                                                 
                    igdumdim32  (???)     GTPIN_IGC_Instrument                                
                           ???  (???)     ???                                                 


-------------------------------

**Comment by rentgen777** [comment]
And me too...

Fri Aug 14 03:50:30 2020


Application has crashed: C++ exception
mods/WolfHUD-master/lua/PrePlanManager.lua:637: attempt to call method '_get_type_cost' (a nil value)



SCRIPT STACK

saved_assets_name() @mods/WolfHUD-master/lua/PrePlanManager.lua:624
@mods/WolfHUD-master/lua/PrePlanManager.lua:174
modify_func() lib/managers/menumanager.lua:7413
get_node() core/lib/managers/menu/coremenudata.lua:36
get_node() core/lib/managers/menu/coremenulogic.lua:310
core/lib/managers/menu/coremenulogic.lua:77
_execute_action_queue() core/lib/managers/menu/coremenulogic.lua:54
update() core/lib/managers/menu/coremenulogic.lua:64
update() core/lib/managers/menu/coremenumanager.lua:155
original() lib/managers/menumanagerpd2.lua:20
update() @mods/base/req/core/Hooks.lua:260
original() lib/setups/setup.lua:858
update() @mods/base/req/core/Hooks.lua:185
update() lib/setups/gamesetup.lua:718
update() lib/setups/networkgamesetup.lua:21
core/lib/setups/coresetup.lua:557


-------------------------------

Callstack:

         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
                           ???  (???)     ???                                                 


-------------------------------

Current thread: Main

-------------------------------

System information:
	Application version : 1.96.906
	CPU : Intel(R) Core(TM) i3-4130 CPU @ 3.40GHz (1 core); SSE; SSE2; SSE3; SSSE3; SSE4.1; SSE4.2
	DirectX : 10.0 
	GPU : NVIDIA GeForce GTX 660 / nvd3dum.dll[27.21.14.5148]
	Language : russian
	Memory :     8129MB 1008KB
	OS : 6.1.7601 (Service Pack 1) 0x100-0x1 (64 bits)
	Physics : threaded
	Renderer : DX9 threaded
	Sound :  (Bluetooth Audio Renderer (Bluetooth СЃС‚РµСЂРµРѕ Р°СѓРґРёРѕ))

**Comment by PhantomGamers** [comment]
Don't need a bunch of people posting the same exact log, if you're getting a different error when crashing post it but otherwise just need to wait for someone to figure out how to fix it

**Comment by K1R1Y4** [comment]
There's actually a bunch of features that just seem to be broken after the last update, like damage icons not working at all and stuff. It's pretty standard for mods to break as new updates come out, so it might take a while untill everything is updated to the game's latest version.

**Comment by PhantomGamers** [comment]
Fixed in #866

**Comment by PhantomGamers** [comment]
#867 is better

**Comment by Socorald** [comment]
Guys, maybe I'm wrong, but these fixes do not work on 199.6

**Comment by K1R1Y4** [comment]
It works for me. This fix was made specifically to fix 199.6, which is the version that caused the issue in the fist place

> Guys, maybe I'm wrong, but these fixes do not work on 199.6

**Comment by K1R1Y4** [comment]
> Guys, maybe I'm wrong, but these fixes do not work on 199.6

This is an issue with 199.6 specifically. And the fix worked just fine for me.

**Comment by fuad-tareq** [comment]
Seems to me like the issue is still happening. Though it didn't trigger for me when I had 0 saved plans (i.e. I was able to click "Load Plans" and it showed me there were 0 without crashing). After I saved a plan though, it started crashing whenever I clicked "Load Plans".

**Comment by PhantomGamers** [comment]
> Seems to me like the issue is still happening. Though it didn't trigger for me when I had 0 saved plans (i.e. I was able to click "Load Plans" and it showed me there were 0 without crashing). After I saved a plan though, it started crashing whenever I clicked "Load Plans".

Yes, we know what the cause is and what the fix is. Refer to the earlier comments & the pull requests.

**Comment by fuad-tareq** [comment]
That's not what ur last comment said.
